pallets/settlement-groth16/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+#[cfg(test)]
+pub mod mock;
+
+#[cfg(any(test, feature = "runtime-benchmarks"))]
+mod dummy_circuit;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+mod data_structures;
+mod groth16;
+mod groth16_generic;
+mod weight;
+
+pub use weight::WeightInfo;
+
+#[frame_support::pallet]
+pub mod pallet {
+
+    use super::WeightInfo;
+    use crate::groth16::{Curve, Groth16, Groth16Error, Proof, Scalar, VerificationKeyWithCurve};
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
+    use hp_poe::OnProofVerified;
+    use sp_core::H256;
+    use sp_io::hashing::keccak_256;
+    use sp_std::vec::Vec;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type OnProofVerified: OnProofVerified;
+        type WeightInfo: WeightInfo;
+        const MAX_NUM_INPUTS: usize;
```

Comment:
You can use constant paramenter : `#[pallet::constant]`

Reply:
Fixed in https://github.com/HorizenLabs/zkVerify/pull/66/commits/222bb7568e108e7fe40ebbb95fcbdd835926f01d